### PR TITLE
Account for recycled water in pen manure cleaning water use calculations

### LINKED ISSUE #1062: Add option to use recycled water for pen manure cleaning
**Is your feature request related to a problem? Please describe.**
The manure module ManureHandler class calculates the amount of water used to clean the pen and adds that to the manure mass. On many farms, most, if not all of this water is recycled from the manure management system so is not 'new' water that is added every day. This causes the total manure volume (especially for systems that use flushing for the pen cleaning) to be unrealistically large. 

**Describe the solution you'd like**
 I would like to add a '`water recycling fraction`' parameter to the manure handler configuration inputs so that we can only add the fraction of water that is not recycled to the manure management system and have a more realistic volume of manure. This '`water recycling fraction`' would be used in the ManureHandler class function `calc_cleaning_water_volume_in_main_barn()`. The equation would be updated to: 
`cleaning_water_volume = num_animals * self.config.cleaning_water_use_rate*(1-water_recycling_fraction)`

**Describe alternatives you've considered**
An alternative would be to update the values for the clearning water use rates directly. This would be easier now but would not allow us to distinguish between those farms that recycle water and those that don't in the future. 

**Additional context**

### PR DESCRIPTION
**Pull request by ew3361zh**
PR to address issue #1062.

Updates:

1. New calculation of `cleaning_water_volume` in `calc_cleaning_water_volume_in_main_barn()` which incorporates the `cleaning_water_recycle_fraction`.
2. Including `cleaning_water_recycle_fraction` in `DefaultManureHandlerConfigFactory` attributes based on rates for each manure handler type shared in issue #1062. Note: these may be updated once Kaitlyn has had a chance to review.
3. Updates this new field `cleaning_water_recycle_fraction` in input files, metadata files, docstrings formatting, and unit testing.

### CODE REVIEW

**Review by allisterakun — APPROVED** [review]
Good job, Niko! Changes are straight-forward, LGTM!

File: RUFAS/routines/manure/manure_handlers/manure_handler_classes.py [hunk 1]
```diff
@@ -239,25 +254,31 @@ class ManureHandlerConfig:
         Number of cleanings per day.
     daily_tillage_frequency : int
         Number of times per day that compost bedding is tilled.
+    cleaning_water_recycle_fraction : float
+        Fraction of water that is not recycled.
```
Comment (KFosterReed) [inline RUFAS/routines/manure/manure_handlers/manure_handler_classes.py]: Can this read: 
'Fraction of cleaning water that is from recycled (not fresh) water sources'

Comment (ew3361zh) [reply RUFAS/routines/manure/manure_handlers/manure_handler_classes.py]: Absolutely

Comment (ew3361zh) [reply RUFAS/routines/manure/manure_handlers/manure_handler_classes.py]: I updated it in the metadata description fields as well.

**Review by KFosterReed — APPROVED** [review]
Looks Great!!

**Comment by KFosterReed** [comment]
This works as expected!

~80 Cow Pen Flush System without recycling: 
default_scenario_3-cleaning-water-volume-10-Jan-2024_Wed_14-15-31
default_scenario_3_handler_man_vol-10-Jan-2024_Wed_14-15-31

~ 80 Cow Pen Flush System With recycling: 
default_scenario_3-cleaning-water-volume-10-Jan-2024_Wed_14-01-29

default_scenario_3_handler_man_vol-10-Jan-2024_Wed_14-01-29


Great work! Thanks Niko :)

**Pull request merged**